src/rpdk/core/contract/suite/contract_asserts.py

```diff
@@ -139,3 +128,34 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return decorator_wrapper
+
+
+@decorate
+def response_contains_resource_model_equal_current_model(
+    resource_client, input_model, output_model
+):
+    # only comparing properties in input model to those in output model and
+    # ignoring extraneous properties that maybe present in output model.
+    assertion_error_message = (
+        "All properties specified in the request MUST "
+        "be present in the model returned, and they MUST"
+        " match exactly, with the exception of properties"
+        " defined as writeOnlyProperties in the resource schema"
+    )
+    output_model = output_model["resourceModel"]
+    try:
+        for key in input_model:
+            if key in resource_client.properties_without_insertion_order:
+                assert test_unordered_list_match(input_model[key], output_model[key])
+            else:
+                assert input_model[key] == output_model[key], assertion_error_message
+    except KeyError as e:
+        raise AssertionError(assertion_error_message) from e
+
+
+def test_unordered_list_match(inputs, outputs):
+    assert len(inputs) == len(outputs)
+    try:
+        assert all(input in outputs for input in inputs)
```

Comment:
maybe instead out `input` and `output` we should be a bit more specific? like handler output keys or something like that

Reply:
this was non_discript for a reason it is being used in multiple places. Like we check input to create handler to output of create handler or input to create handler with output of read handler.